src/librbd/Operations.cc

```diff
@@ -1637,6 +1638,49 @@ void Operations<I>::execute_migrate(ProgressContext &prog_ctx,
   req->send();
 }
 
+template <typename I>
+int Operations<I>::sparsify(size_t sparse_size, ProgressContext &prog_ctx) {
+  CephContext *cct = m_image_ctx.cct;
+  ldout(cct, 20) << "sparsify" << dendl;
```

Comment:
It might be a good idea to validate the sparse size is (1) a power-of-two and (2) is greater then some lower bound (4KiB?) and less than the object size.